### ISSUE TITLE
Updates, NPM, and Instance Mode

### ISSUE THREAD

**Issue by JackWH** [freshfork/p5.EasyCam#26]
Hi! 👋 Is this library still maintained? I've been unable to import it (cleanly, anyway) into a Yarn/TypeScript project, as no package is published on npm.

Furthermore, when I tried to copy the source code in manually, I couldn't get it to work with [p5's Instance Mode](https://github.com/processing/p5.js/wiki/Global-and-instance-mode). The library seems to assume p5 is always available globally, which is rarely the case in modern JS projects.

Thanks!

**Comment by jwdunn1** [comment]
Yes, this library in maintained. 1) Coincidentally, a new NPM package was [published](https://www.npmjs.com/package/p5.easycam) this morning. 2) The TypeScript issue #13 is still under review with a testable version of it being merged into the repo (after the npm package was published). 3) Yes, the p5.EasyCam library works in instance mode: [example](https://freshfork.github.io/p5.EasyCam/examples/MultiCanvas)

**Comment by JackWH** [comment]
Thanks @jwdunn1 for the speedy reply 👍 The instance mode issue is a little different, I've added some clarification below.

The npm package doesn't `export` any methods, so it can't be imported as a module. Instead the plugin has to be loaded into the global scope. In instance mode p5 won't always be available in the global scope, and the `p5` class also might be imported under an alias.

[These lines](https://github.com/freshfork/p5.EasyCam/blob/9782964680f6a5c4c9bee825c475d9f2021d5134/p5.easycam.js#L1677-L1685) seem to be causing the problem. They assume a `p5` variable is available at runtime. Even if the p5 runtime is loaded and available, hard-coding onto the `p5.prototype` prevents using this plugin in a class which has a p5 instance assigned to a property. More info [here](https://github.com/processing/p5.js/issues/3813).

If you'd welcome a PR for this, I can have a go at it! 😊

**Comment by jwdunn1** [comment]
Fascinating! Thanks for your insights. Yes, contributions are welcome!

**Comment by JackWH** [comment]
I got a prototype working last night, just need to make sure it stays backwards-compatible in the global context. Will get a PR up when I get some time to come back to this over the next few days. Thanks!

**Comment by jwdunn1** [comment]
@JackWH Were you able to import the modified version (from the PR) successfully into your Yarn/TypeScript project?
